packages/aws-cdk-lib/aws-ses/test/configuration-set.test.ts

```diff
@@ -40,3 +40,57 @@ test('configuration set with options', () => {
     },
   });
 });
+
+test('configuration set with both engagement metrics and optimized shared delivery', () => {
+  new ConfigurationSet(stack, 'ConfigurationSet', {
+    vdmOptions: {
+      engagementMetrics: true,
+      optimizedSharedDelivery: true,
+    },
+  });
+
+  Template.fromStack(stack).hasResourceProperties('AWS::SES::ConfigurationSet', {
+    VdmOptions: {
+      DashboardOptions: {
+        EngagementMetrics: 'ENABLED',
+      },
+      GuardianOptions: {
+        OptimizedSharedDelivery: 'ENABLED',
+      },
+    },
+  });
+});
+
+test('configuration set with engagement metrics only', () => {
+  new ConfigurationSet(stack, 'ConfigurationSet', {
+    vdmOptions: {
+      engagementMetrics: true,
+    },
+  });
+
+  Template.fromStack(stack).hasResourceProperties('AWS::SES::ConfigurationSet', {
+    VdmOptions: {
+      DashboardOptions: {
+        EngagementMetrics: 'ENABLED',
+      },
+      GuardianOptions: Match.absent(),
+    },
+  });
+});
+
+test('configuration set with optimized shared delivery only', () => {
+  new ConfigurationSet(stack, 'ConfigurationSet', {
+    vdmOptions: {
+      optimizedSharedDelivery: true,
+    },
+  });
+
+  Template.fromStack(stack).hasResourceProperties('AWS::SES::ConfigurationSet', {
+    VdmOptions: {
+      DashboardOptions: Match.absent(),
+      GuardianOptions: {
+        OptimizedSharedDelivery: 'ENABLED',
+      },
+    },
+  });
+});
```

Comment:
These tests seem a bit superfluous, given they're already being covered by the first one. It'd be more appropriate to test `vdmOptions: {(false, false}` and `vdmOptions: {}`,

Reply:
I've added test cases for when both options in vdmOptions are set to "DISABLED" and when vdmOptions itself is not configured. 

Additionally, I'm sorry to say that I've realized from the results of these tests that the "DISABLED" setting cannot be implemented. Therefore, I'm revisiting the implementation of vdmOptions.

Reply:
Have you tried setting it to `DISABLED` with `ENABLED` account wide settings? That might be the only allowed use case

Reply:
Yes, I did and I could.

As I mentioned in a separate comment, the configuration set level settings can be deployed on their own, but to enable VDM, it is appropriate to configure them together with the account level settings.
Therefore, I also added the account level settings to the integration tests.